mdagent/tools/base_tools/preprocess_tools/uniprot.py

```diff
@@ -0,0 +1,1250 @@
+            polling_interval: The interval to poll the API for results.
+                Defaults to 3 seconds.
+
+        Returns:
+            A list of mapped database entries from the 'to' field if
+                successful, otherwise an empty list.
+        """
+        with requests.Session() as session:
+            session.mount(
+                "https://",
+                HTTPAdapter(
+                    max_retries=Retry(
+                        total=5,
+                        backoff_factor=0.25,
+                        status_forcelist=[500, 502, 503, 504],
+                    )
+                ),
+            )
+            response = session.post(
+                f"{self.API_URL}/idmapping/run",
+                data={"from": from_db, "to": to_db, "ids": query},
+            )
+            response.raise_for_status()
+            job_id = response.json()["jobId"]
+
+            while True:
+                response = session.get(f"{self.API_URL}/idmapping/status/{job_id}")
+                response.raise_for_status()
+                status_data = response.json()
+                if status_data.get("jobStatus") == "RUNNING":
+                    print(f"Job is running. Retrying in {polling_interval}s.")
+                    time.sleep(polling_interval)
+                else:
+                    break
+
+            response = session.get(f"{self.API_URL}/idmapping/details/{job_id}")
+            response.raise_for_status()
+            results_link = response.json().get("redirectURL")
+
+            response = session.get(results_link)
+            response.raise_for_status()
+            if response.headers["Content-Type"] != "application/json":
+                raise ValueError("Expected JSON response but got a different format.")
+
+            results_json = response.json()
+            results = results_json.get("results", [])
+            return [r["to"] for r in results]
+
+    def get_data(
+        self, query: str, desired_field: str, format_type: str = "json"
+    ) -> list | None:
+        """
+        Helper function to get data from the Uniprot API.
+
+        Args:
+            query: The query string to search (e.g. 'hemoglobin')
+            desired_field: The desired field to retrieve from the API
+            format_type: The format of the data to retrieve. Defaults to 'json'.
+
+        Returns:
+            The data retrieved from the API or None if no data is found.
+        """
+        with requests.Session() as session:
+            session.mount(
+                "https://",
+                HTTPAdapter(
+                    max_retries=Retry(
+                        total=5,
+                        backoff_factor=0.25,
+                        status_forcelist=[500, 502, 503, 504],
+                    )
+                ),
+            )
+            url = f"https://rest.uniprot.org/uniprotkb/search?fields={desired_field}&format={format_type}&query={query}"
```

Comment:
whoops! bit too long 
```suggestion
            url = f"https://rest.uniprot.org/uniprotkb/search?fields={desired_field}\
            &format={format_type}&query={query}"
```

Reply:
I'd rather not break it up for readability purposes since its a URL